### PR TITLE
Add favicon

### PR DESCRIPTION
**Pull request by rmuir**
Looks like this. Let the bikeshedding begin :)

favicon

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/http/HttpServer.java [hunk 1]
```diff
@@ -131,6 +137,22 @@ public void process(RestRequest request, RestChannel channel, RestFilterChain fi
         }
     }
 
+    void handleFavicon(HttpRequest request, HttpChannel channel) {
+        if (request.method() == RestRequest.Method.GET) {
+            try {
+                try (InputStream stream = getClass().getResourceAsStream("/config/favicon.ico")) {
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/http/HttpServer.java]: should we keep this in memory for performance reasons?

**Comment by s1monw** [comment]
LGTM I raised one little concern :8ball:

File: core/src/main/java/org/elasticsearch/http/HttpServer.java (hunk 1 above)
Comment (rmuir) [inline core/src/main/java/org/elasticsearch/http/HttpServer.java]: i dunno, it only impacts users using the browser. i figure that isnt perf critical, the server doesn't get hammered with those. maybe i'm wrong. file is 1KB.

Comment (s1monw) [inline core/src/main/java/org/elasticsearch/http/HttpServer.java]: I can't come up with any more seriousness :)

**Comment by tlrx** [comment]
Nice... Looks like you were bored with unicast/multicast things :)

**Comment by jpountz** [comment]
LGTM

**Pull request merged**